settlement/dymension/dymension.go

```diff
@@ -350,6 +350,39 @@ func (d *HubClient) GetBatchAtIndex(rollappID string, index uint64) (*settlement
 	return d.convertStateInfoToResultRetrieveBatch(&stateInfoResp.StateInfo)
 }
 
+func (d *HubClient) GetHeightState(rollappID string, h uint64) (*settlement.ResultGetHeightState, error) {
+	// TODO: dry out with GetBatchAtIndex
+	var stateInfoResp *rollapptypes.QueryGetStateInfoResponse
+	err := d.RunWithRetry(func() error {
+		var err error
+		stateInfoResp, err = d.rollappQueryClient.StateInfo(d.ctx,
+			&rollapptypes.QueryGetStateInfoRequest{RollappId: d.config.RollappID, Height: h})
+
+		if status.Code(err) == codes.NotFound {
+			return retry.Unrecoverable(settlement.ErrBatchNotFound)
```

Comment:
we cant return an error here because this situation is dfeinitely feasibale :
1. we synced from p2p until height 100
2. sequencer only amange to write 95 to SL
3. we restarted, serarching from 100 and it doesn't exist yet.